make reset private and remove from Atom interface

dagda1 opened the pull request:
As discussed, remove `reset` from the `Atom` interface and access is now from a non-public `_reset`.

E.g. in `helpers` in `@bigtest/server` which is the only usage:

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
  (actions.atom as any)._reset((initial: OrchestratorState) => ({ ...initial, manifest, bundler }));
  ```

Commit:
make reset private and remove from Atom interface

cowboyd commented on packages/server/test/helpers.ts:
```diff
@@ -124,7 +124,8 @@ after(async function() {
 
 beforeEach(() => {
-  actions.atom.reset(initial => ({ ...initial, manifest, bundler }));
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (actions.atom as any)._reset((initial: OrchestratorState) => ({ ...initial, manifest, bundler }));
```
Can we perhaps extract this into a test helper that comes with the atom package? Like `resetAtom()` and then invoke it from here?

dagda1 replied:
@cowboyd I like that idea a lot.  On its way

Commit:
export a resetAtom function

cowboyd reviewed (approved):
nice! Optional: should we add a comment to `resetAtom` that this should be used for testing purposes only?

Commit:
remove resetAtom export from @bigtest/atom main

dagda1 commented:
> nice! Optional: should we add a comment to `resetAtom` that this should be used for testing purposes only?

I've added the comment and removed the export from `src/index.ts` so you have to go to the `dist` directory to get it if you want it.

```ts
import { resetAtom } from '@bigtest/atom/dist/atom';
```
